Source/WebCore/rendering/style/StyleGridData.h

```diff
@@ -117,14 +117,34 @@ class StyleGridData : public RefCounted<StyleGridData> {
     const unsigned& autoRepeatColumnsInsertionPoint() const { return m_autoRepeatColumnsInsertionPoint; }
     const unsigned& autoRepeatRowsInsertionPoint() const { return m_autoRepeatRowsInsertionPoint; }
 
-    const AutoRepeatType& autoRepeatColumnsType() const { return m_autoRepeatColumnsType; }
-    const AutoRepeatType& autoRepeatRowsType() const { return m_autoRepeatRowsType; }
-
     const bool& subgridRows() const { return m_subgridRows; };
     const bool& subgridColumns() const { return m_subgridColumns; }
 
+    // Masonry Spec Section 2
+    // "If masonry is specified for both grid-template-columns and grid-template-rows, then the used value for grid-template-columns is none, 
+    // and thus the inline axis will be the grid axis."
+
     bool masonryRows() const { return m_masonryRows; }
-    bool masonryColumns() const { return m_masonryColumns; }
+    bool masonryColumns() const { return !m_masonryRows && m_masonryColumns; }
```

Comment:
This will change what we serialise the computed value for `grid-template-columns`, right?

There's precedent for adding a `usedMasonryColumns` getter for cases where the computed value and used value aren't the same (see `usedZIndex` and `usedTransformStyle3D`)